Assets/MixedRealityToolkit/_Core/Devices/OpenVR/OpenVRDeviceManager.cs

```diff
@@ -206,29 +206,11 @@ private GenericOpenVRController GetOrAddController(XRNodeState xrNodeState)
             var pointers = RequestPointers(controllerType, controllingHand);
             var inputSource = InputSystem?.RequestNewGenericInputSource($"{CurrentControllerType} Controller {controllingHand}", pointers);
 
-            GenericOpenVRController detectedController = null;
-
-            switch (CurrentControllerType)
-            {
-                case SupportedControllerType.GenericOpenVR:
-                    detectedController = new GenericOpenVRController(TrackingState.NotTracked, controllingHand, inputSource);
-                    break;
-                case SupportedControllerType.ViveWand:
-                    detectedController = new ViveWandController(TrackingState.NotTracked, controllingHand, inputSource);
-                    break;
-                case SupportedControllerType.ViveKnuckles:
-                    detectedController = new ViveKnucklesController(TrackingState.NotTracked, controllingHand, inputSource);
-                    break;
-                case SupportedControllerType.OculusTouch:
-                    detectedController = new OculusTouchController(TrackingState.NotTracked, controllingHand, inputSource);
-                    break;
-                case SupportedControllerType.OculusRemote:
-                    detectedController = new OculusRemoteController(TrackingState.NotTracked, controllingHand, inputSource);
-                    break;
-            }
+            GenericOpenVRController detectedController = Activator.CreateInstance(controllerType, 
+                new object[] { TrackingState.NotTracked, controllingHand, inputSource }) as GenericOpenVRController;
 
             Debug.Assert(detectedController != null);
-            detectedController?.SetupConfiguration(controllerType);
+            detectedController.SetupConfiguration(controllerType);
```

Comment:
Let's keep the null check here.

Even if unity throws the assert, it'll still crash unfortunately :/